### PR TITLE
[Feature][E2E] Setup E2E under JDK 11

### LINKED ISSUE #15267: [Feature][E2E] Setup E2E under JDK 11
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

_No response_

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by pegasas**
## Purpose of the pull request

resolve: https://github.com/apache/dolphinscheduler/issues/15267

## Brief change log

setup jdk11 for e2e

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -56,6 +59,11 @@ jobs:
         uses: ./.github/actions/sanity-check
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v2
+        with:
+          java-version: ${{ matrix.java }}
+          distribution: 'adopt'
```
Comment (SbloodyS) [inline .github/workflows/e2e.yml]: This only setup jdk of the host. You should also modify the runtime environment of e2e containers at the same time.

Comment (pegasas) [reply .github/workflows/e2e.yml]: Thanks @SbloodyS !
I've updated the PR.

File: .github/workflows/e2e-k8s.yml [hunk 2]
```diff
@@ -47,11 +47,19 @@ jobs:
     needs: paths-filter
     if: ${{ (needs.paths-filter.outputs.not-ignore == 'true') || (github.event_name == 'push') }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ '8', '11' ]
```
Comment (Gallardot) [inline .github/workflows/e2e-k8s.yml:52]: Adding JDK 11 to this job might not have much effect, as the current Docker image only supports JDK 8. Unless we also build a Docker image for JDK 11. We welcome your contribution of a JDK 11 image.

Comment (SbloodyS) [reply .github/workflows/e2e-k8s.yml:52]: > Adding JDK 11 to this job might not have much effect, as the current Docker image only supports JDK 8. Unless we also build a Docker image for JDK 11. We welcome your contribution of a JDK 11 image.

This is as the same as e2e.

Comment (pegasas) [reply .github/workflows/e2e-k8s.yml:52]: It seems we directly use mvnw for building image, which may not support JDK11.
I found that there is docker CLI in below steps.
How about we use mvnw for building jar, use Dockerfile for building image?
Is there any policy in apache against base docker image?

Comment (Gallardot) [reply .github/workflows/e2e-k8s.yml:52]: > It seems we directly use mvnw for building image, which may not support JDK11. I found that there is docker CLI in below steps. How about we use mvnw for building jar, use Dockerfile for building image? Is there any policy in apache against base docker image?

`mvnw` executes Docker CLI commands through a plugin.

There are two stages here. The host machine needs to be in a JDK11 environment when executing `mvnw` to build the jar file. When building the Docker image, the base Docker image also needs to be JDK11.

Comment (pegasas) [reply .github/workflows/e2e-k8s.yml:52]: make sense. I will diginto `mvnw` base docker image setup, then.

File: .github/workflows/e2e-k8s.yml [hunk 3]
```diff
@@ -61,12 +69,20 @@ jobs:
...
           -pl dolphinscheduler-master \
           -pl dolphinscheduler-worker -am
+      - name: Build Image
+        run: |
+          components=("master" "worker" "api" "tools" "alert-server")
+          for component in "${components[@]}"; do
+            docker build --build-arg JAVA_VERSION=${{ matrix.java }} \
+            -t apache/dolphinscheduler-${component}:ci \
+            -f dolphinscheduler-${component}/src/main/docker/Dockerfile
+          done
```
Comment (Gallardot) [inline .github/workflows/e2e-k8s.yml]: Please take a look this   [plugin](https://github.com/apache/dolphinscheduler/blob/ce11674668cf26747f34c3474763c32e1ba5b6d1/pom.xml#L455-L517).

You can add a new environment variable for the Java version in the plugin and pass it to the Docker CLI.

**Review by Gallardot — CHANGES_REQUESTED** [review]

**Pull request closed without merging**